modules/database/test/ioc/db/jlinkz.c

```diff
@@ -129,7 +129,7 @@ long z_putval(struct link *plink, short dbrType,
     if(nRequest==0) return 0;
 
     epicsMutexLock(priv->lock);
-    ret = (*pconv)(&priv->value, pbuffer, NULL);
+    ret = pconv(pbuffer, &priv->value, NULL);
```

Comment:
Core group: Arg order change, is this right?

Reply:
I believe my change here is correct, but incomplete. This routine doesn't actually get called by the `dbPutLinkTest` program at all, so we've never noticed that the code has a problem. The `lset::putValue()` method would be used by a `dbPutLink()` call in a record to send the `dbrType` value at `*pbuffer` through a `DBF_OUTLINK`, but we currently only test with input links and the `xRecord.c` doesn't call `dbPutLink()` anywhere.

For this link type `z_putval()` is trying to write the `*pbuffer` value to its own `epicsInt32 zpriv::value` member. However `dbFastPutConvertRoutine[DBF_LONG][dbrType]` looks up the function that converts from `DBF_LONG` to `dbrType` — both the Get and Put arrays use the order `[from][to]`, as do the functions arguments. The original code here looks to have been copied from the `z_getval()` method, but only the array used to look up the convert routine was changed.

New commits correct this routine, add tests, and provide some documentation.